xray/usage/reportusage.go

```diff
@@ -0,0 +1,177 @@
+package usage
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	versionutil "github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/http/httpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray"
+)
+
+const (
+	minXrayVersion = "3.80.0"
+
+	xrayUsageApiPath = "api/v1/usage/events/send"
+
+	EcosystemUsageBaseUrl = "https://usage-ecosystem.jfrog.info/"
+	ecosystemUsageApiPath = "api/usage/report"
+
+	ReportUsagePrefix = "Usage Report: "
+)
+
+type ReportUsageAttribute struct {
+	AttributeName  string
+	AttributeValue string
+}
+
+func (rua *ReportUsageAttribute) isEmpty() bool {
+	return rua.AttributeName == ""
+}
+
+type ReportXrayEventData struct {
+	Attributes map[string]string `json:"data,omitempty"`
+	ProductId  string            `json:"product_name"`
+	EventId    string            `json:"event_name"`
+	Origin     string            `json:"origin,omitempty"`
+}
+
+func SendXrayUsageEvents(serviceManager xray.XrayServicesManager, events ...ReportXrayEventData) error {
+	if len(events) == 0 {
+		return errorutils.CheckErrorf(ReportUsagePrefix + "Nothing to send.")
+	}
+	config := serviceManager.Config()
+	if config == nil {
+		return errorutils.CheckErrorf(ReportUsagePrefix + "Expected full config, but no configuration exists.")
+	}
+	xrDetails := config.GetServiceDetails()
+	if xrDetails == nil {
+		return errorutils.CheckErrorf(ReportUsagePrefix + "Xray details not configured.")
+	}
+	xrayVersion, err := xrDetails.GetVersion()
```

Comment:
the serviceManager can get this method, since we already using it I don't see any point adding another argument for xray version, it will make the usage more complicated.